compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -862,19 +865,15 @@ impl CrateInfo {
             is_no_builtins: Default::default(),
             native_libraries: Default::default(),
             used_libraries: tcx.native_libraries(LOCAL_CRATE).iter().map(Into::into).collect(),
-            crate_name: Default::default(),
+            crate_name: UnordMap::with_capacity(n_crates),
             used_crates,
-            used_crate_source: Default::default(),
+            used_crate_source: UnordMap::with_capacity(n_crates),
             dependency_formats: tcx.dependency_formats(()).clone(),
             windows_subsystem,
             natvis_debugger_visualizers: Default::default(),
         };
-        let crates = tcx.crates(());
 
-        let n_crates = crates.len();
         info.native_libraries.reserve(n_crates);
-        info.crate_name.reserve(n_crates);
-        info.used_crate_source.reserve(n_crates);
```

Comment:
no function or associated item named `with_capacity` found for struct `indexmap::map::IndexMap`.

You mean change `native_libraries` to use `UnordMap`? it seems two references are using `iter()` on `native_libraries`, changing to use `UnordMap` may change the behavior.

Reply:
Strange, I just assumed it would have a `with_capacity` method like most collections. This is fine to leave as is, then, I wouldn't change the type if you don't have to.

Reply:
(It doesn't have `with_capacity` because that function requires the default RandomState hasher; you'd have to do `IndexMap::with_capacity_and_hasher(cap, Default::default())`. No different than using the std HashMap with a custom hasher.)